docs/docs/user/references/components/kafka-app.md

```diff
@@ -0,0 +1,41 @@
+# KafkaApp
+
+Subclass of [_KubernetesApp_](kubernetes-app.md).
+
+### Usage
+
+- Defines a [streams-bootstrap](https://github.com/bakdata/streams-bootstrap#usage){target=_blank} component
+- Should not be used in `pipeline.yaml` as the component can be defined as either a [StreamsApp](streams-app.md)
+  or a [ProducerApp](producer-app.md)
+- Often used in `defaults.yaml`
+
+### Configuration
+
+??? example "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/kafka-app.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+In addition to [KubernetesApp's `deploy`](kubernetes-app.md#deploy):
+
+- Create topics if provided (optional)
+- Submit Avro schemas to the registry if provided (optional)
+
+#### destroy
+
+Uninstall Helm release.
+
+#### reset
+
+Do nothing.
+
+#### clean
+
+Do nothing.
```

Comment:
do you think do nothing is a good expression for this ? maybe something like "not implemented" would be better?

Reply:
I remember the first time we wrote the docs we used the term `not implemented` and this sounds IMO that it is still a work in progress and needs to be implemented. WDYT? @irux